### PR TITLE
[RHPAM-2175] Set login-modules to sufficient

### PR DESCRIPTION
**Pull request by ruromero**
Signed-off-by: Ruben Romero Montes <rromerom@redhat.com>

Fix https://issues.jboss.org/browse/RHPAM-2175

### CODE REVIEW

File: jboss-kie-wildfly-common/added/launch/jboss-kie-wildfly-security-login-modules.sh [hunk 1]
```diff
@@ -67,7 +67,7 @@ function configure_ldap_login_module() {
         return
     fi
     log_info "AUTH_LDAP_URL is set to ${AUTH_LDAP_URL}. Added LdapExtended login-module"
-    local login_module=$(build_login_module "LdapExtended" "required")
+    local login_module=$(build_login_module "LdapExtended" "sufficient")
```
Comment (spolti) [inline jboss-kie-wildfly-common/added/launch/jboss-kie-wildfly-security-login-modules.sh]: shouldn't it be configurable and defaulted to sufficient?

File: jboss-kie-wildfly-common/added/launch/jboss-kie-wildfly-security-login-modules.sh [hunk 2]
```diff
@@ -94,7 +94,7 @@ function configure_ldap_login_module() {
 
     # RHPAM-1422, if the RealmDirect is set as Required, ldap auth will fail.
     # TODO remove it out as part of the CLOUD-2750
-    sed -i 's|<login-module code="RealmDirect" flag="required">|<login-module code="RealmDirect" flag="optional">|' "${CONFIG_FILE}"
+    sed -i 's|<login-module code="RealmDirect" flag="required">|<login-module code="RealmDirect" flag="sufficient">|' "${CONFIG_FILE}"
```
Comment (spolti) [inline jboss-kie-wildfly-common/added/launch/jboss-kie-wildfly-security-login-modules.sh]: same than above.

**Comment by ruromero** [comment]
@spolti let me know if this is what you were referring to. Thanks

**Comment by ruromero** [comment]
Needs to be set by the user. By default is not set and both configs will be set as `sufficient`

**Comment by ruromero** [comment]
@spolti Just to clarify, login modules will be set as `sufficient` if SSO_URL or LDAP_URL are used. Otherwise the default login module will be set as `required`

**Comment by spolti** [comment]
@ruromero to you mind to document the new env on the needed image.yaml files?

**Comment by ruromero** [comment]
@spolti the suggested change does not affect this PR. Is there anything else that needs to be changed?
Thanks

**Comment by spolti** [comment]
@ruromero Sury for the delay, you missed to apply the same changes on rhdm images. Once this PR is in place we can go ahead and merge the PRs.

Thanks.

**Comment by ruromero** [comment]
Depends on #348 
Rebase and keep only EXTERNAL_AUTH_ONLY changes

**Comment by spolti** [comment]
https://github.com/jboss-container-images/jboss-kie-modules/pull/348 is merged. Ruben can you please rebase?

**Comment by ruromero** [comment]
@spolti rebased, please review

**Pull request merged**